core/src/main/scala/algebra/Eq.scala

```diff
@@ -70,6 +70,19 @@ object Eq extends EqFunctions {
     }
 
   /**
+   * Create an `Eq` instance that defines equality via the conjunction
+   * of each of the given functions `fs`.
+   */
+  def fromAll[@sp A](fs : ((A,A) => Boolean)*) : Eq[A] =
+    new Eq[A] {
+      def eqv(x: A, y: A) =
+        fs.foldLeft(true) {
```

Comment:
is this what we intend? This is different than the logic you had before (the others below are not).

Here you are looking for the first true, before you were keeping if all were true. I think we want the all true case.